controllers/blockdevice/blockdevice_controller.go

```diff
@@ -145,9 +119,16 @@ func (r *ReconcileBlockDevice) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileBlockDevice) updateBDStatus(state openebsv1alpha1.DeviceClaimState, instance *openebsv1alpha1.BlockDevice) error {
+func (r *BlockDeviceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&apis.BlockDevice{}).
+		Owns(&apis.BlockDevice{}).
+		Complete(r)
```

Comment:
What is the use of this function?

Reply:
It is equivalent to Build() functions we have in builder pattern